codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketClientProtocolConfig.java

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright 2019 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http.websocketx;
+
+import io.netty.handler.codec.http.EmptyHttpHeaders;
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.handler.codec.http.websocketx.WebSocketClientProtocolHandler.ClientHandshakeStateEvent;
+import io.netty.util.internal.ObjectUtil;
+
+import java.net.URI;
+
+import static io.netty.util.internal.ObjectUtil.checkPositive;
+
+/**
+ * WebSocket server configuration.
+ */
+public final class WebSocketClientProtocolConfig {
+
+    static final WebSocketClientProtocolConfig DEFAULT = new WebSocketClientProtocolConfig(
+        URI.create("https://localhost/"), null, WebSocketVersion.V13, false,
+        EmptyHttpHeaders.INSTANCE, 65536, true, false, true, true, 10000L, -1, false);
+
+    private final URI webSocketUri;
+    private final String subprotocol;
+    private final WebSocketVersion version;
+    private final boolean allowExtensions;
+    private final HttpHeaders customHeaders;
+    private final int maxFramePayloadLength;
+    private final boolean performMasking;
+    private final boolean allowMaskMismatch;
+    private final boolean handleCloseFrames;
+    private final boolean dropPongFrames;
+    private final long handshakeTimeoutMillis;
+    private final long forceCloseTimeoutMillis;
+    private final boolean absoluteUpgradeUrl;
+
+    private WebSocketClientProtocolConfig(
```

Comment:
Since this is a private constructor, this could be simplified by passing in the Builder as the single argument.

Reply:
it is impossible (would not compile):

a) builder already requires a single parameter - config,
b) default config is defined explicitly with all arguments, see lines 33-35.